action.yml

```diff
@@ -316,6 +332,28 @@ runs:
           digger-cache
       if: ${{ inputs.cache-dependencies == 'true' }}
 
+    - name: restore-s3-cache
+      shell: bash
+      run: |
+        BUCKET=$(echo "${{ inputs.cache-dependencies-s3-bucket }}" | sed 's|^s3://||')
+        REGION="${{ inputs.cache-dependencies-s3-region }}"
+        
+        if [ ! -d "$TF_PLUGIN_CACHE_DIR" ]; then
+          echo "Creating cache directory: $TF_PLUGIN_CACHE_DIR"
+          mkdir -p "$TF_PLUGIN_CACHE_DIR"
+        fi
+        
+        echo "Restoring cache from S3..."
```

Comment:
this is a nice to have but given there is some validation logic and so on in these scripts I wonder if we can isolate both in something like scripts/s3-cache-upload.bash and scripts/s3-cache-download.bash so that we can add a few tests to the scripts independently